[SPARK-125] Handle and test for two edge cases for confused clients

Padawan00123 opened the pull request.

Padawan00123 committed:
[SPARK-125] Build tests for SPARK-125

Padawan00123 committed:
[SPARK-125] Don't use error handling for tests for SPARK-125

Padawan00123 committed:
[SPARK-125] Handle and test for two edge cases that aren't handled in Mecha2

theunkn0wn1 commented on tests/unit/test_ratmama.py:
```diff
@@ -178,6 +206,24 @@ async def test_manual_signal(bot_fx,
+        "unit_test",
+        "Ratsignal some_service")
+
+    monkeypatch.setattr(context, "reply", async_callable_fx)
```
[question] why was the reply attribute patched here? I don't see any usage of `async_callable_fx` used in an assertion as a result of this patch?

theunkn0wn1 commented on src/packages/ratmama/ratmama_parser.py:
```diff
@@ -108,6 +109,16 @@ async def handle_ratmama_announcement(ctx: Context) -> None:
+    client = await User.from_pydle(ctx.bot, client_name)
+
+    if client is not None:
+        client_hostname = client.hostname
```
[Error] Redundant variable assignment

You needlessly assign an instance attribute of a dataclass to a local name (read only, cheap retrieval) , it is better NOT to extract that one attribute from the object, as it may lead to broken expectations.

theunkn0wn1 commented on src/packages/ratmama/ratmama_parser.py:
```diff
@@ -108,6 +109,16 @@ async def handle_ratmama_announcement(ctx: Context) -> None:
 
+    client = await User.from_pydle(ctx.bot, client_name)
+
+    if client is not None:
+        client_hostname = client.hostname
+
+        if client_hostname in (
+                "services.fuelrats.com",
+                "bot.fuelrats.com"):
+            return
```
```suggestion
    if client and client.hostname in ("services.fuelrats.com", "bot.fuelrats.com"):
        logger.warning("attempted to create rescue for a service!")
        # emit a warning to the dispatch, so they can handle this themselves.
        return await ctx.reply("signal attempted to create rescue for a service. Dispatch: please manually inject!")
```

theunkn0wn1 reviewed (changes requested):
Looks ok, but we need to warn the dispatch when we fail in this mode so they can handle it themselve.

Padawan00123 committed:
[SPARK-125] Address code review comments.

This commit removes an unused async_callable_fx, and an unused variable assignment.  We also warn dispatch if a user attempted to create a case for a service, and test for that message.

theunkn0wn1 reviewed (dismissed):
LGTM :+1:

theunkn0wn1 reviewed (approved):
LGTM :+1:
